Tools/clinic/clinic.py

```diff
@@ -5732,8 +5748,17 @@ def run_clinic(parser: argparse.ArgumentParser, ns: argparse.Namespace) -> None:
 def main(argv: list[str] | None = None) -> NoReturn:
     parser = create_cli()
     args = parser.parse_args(argv)
-    run_clinic(parser, args)
-    sys.exit(0)
+    try:
+        run_clinic(parser, args)
+    except ClinicError as exc:
+        msg = textwrap.dedent(f"""\
+            Error in file {exc.filename!r} on line {exc.lineno}:
+            {exc}
+        """)
```

Comment:
pyflakes points out that the `msg` variable here is assigned to but never used

Reply:
Now, this leads down an interesting rabbit hole. `Clinic.parse` is actually preventing `ClinicError` from appearing in `main`; there's an `except Exception` in there. Suggesting the following:

```diff
diff --git a/Tools/clinic/clinic.py b/Tools/clinic/clinic.py
index de621b1505..6fbb395fdd 100755
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -2249,11 +2249,7 @@ def parse(self, input: str) -> str:
                     assert dsl_name in parsers, f"No parser to handle {dsl_name!r} block."
                     self.parsers[dsl_name] = parsers[dsl_name](self)
                 parser = self.parsers[dsl_name]
-                try:
-                    parser.parse(block)
-                except Exception:
-                    fail('Exception raised during parsing:\n' +
-                         traceback.format_exc().rstrip())
+                parser.parse(block)
             printer.print_block(block)
 
         # these are destinations not buffers
@@ -5737,7 +5733,7 @@ def main(argv: list[str] | None = None) -> NoReturn:
             Error in file {exc.filename!r} on line {exc.lineno}:
             {exc}
         """)
-        sys.stderr.write(str(exc))
+        sys.stderr.write(msg)
         sys.exit(1)
     else:
         sys.exit(0)
```

Reply:
That looks promising!

Reply:
There's another catch-all handler in `parse_parameter`. Let's add the tests firsts; there are too many uncovered paths here.